### DISCUSSION TITLE
Cookies in Unraid

### DISCUSSION THREAD

**Discussion by Hawkins1212** [Tzahi12345/YoutubeDL-Material#445] [Q&A] [answered]
Trying to set up to pull from Discovery Plus.  I know this requires a "cookies" file which I have.  I installed docker (on Unraid), went to "Advanced" and enabled "Use Cookies".  For the Set Cookies, i pulled in the cookies.txt file I created.  When trying to DL, I get an error that only registered users are allowed.  Any success with D+?  See below from the logs:

2021-09-11T20:37:12.259Z ERROR: failed to parse for urls starting with https://www.discoveryplus.com/video/[VideoName]

2021-09-11T20:37:15.119Z ERROR: Error while retrieving info on video with URL https://www.discoveryplus.com/video/[VideoName] with the following message: Error: Command failed with exit code 1: /app/node_modules/youtube-dl/bin/youtube-dl --dump-json -f best -o video/%(title)s.mp4 --write-info-json --print-json --write-thumbnail -- https://www.discoveryplus.com/video/[VideoName]

2021-09-11T20:37:15.119Z ERROR: ERROR: This video is only available for registered users. You may want to use --cookies.

**Comment by GlassedSilver** [comment, marked as answer]
Are you using the latest `nightly`? Sounds like you're still on the last release channel build which is WAY old.

![image](https://user-images.githubusercontent.com/1912133/132963647-e92919a6-52e1-49ba-917f-9fcebd4abb3e.png)

This is how your template should look like.

Once you're moved over to the `nightly` branch you can set cookies (and I think they will be automatically applied? @Tzahi12345) by going to **Settings > Advanced > Use Cookies** - TICK & submit cookie file in here as well

If this doesn't help I'm afraid you might be looking at a yt-dl issue more than a specific youtubedl-material or youtubedl-material on unRAID issue.

@Tzahi12345 might be able to help you figure out the exact real where to look for solutions, personally I'm not a Discovery+ user so my scope of being able to help here is limited.


**Edit:** since you may have missed it, here's further advise for fellow unRAIDers using this application ;)
https://github.com/Tzahi12345/YoutubeDL-Material/wiki/unRAID

**Reply by Hawkins1212** [reply]
> Are you using the latest `nightly`? Sounds like you're still on the last release channel build which is WAY old.
> 
> ![image](https://user-images.githubusercontent.com/1912133/132963647-e92919a6-52e1-49ba-917f-9fcebd4abb3e.png)
> 
> This is how your template should look like.
> 
> Once you're moved over to the `nightly` branch you can set cookies (and I think they will be automatically applied? @Tzahi12345) by going to **Settings > Advanced > Use Cookies** - TICK & submit cookie file in here as well

Thanks for this.  I was night using the latest `nightly` so I updated accordingly. I did as you mentioned (ticked, submitted, uploaded) and still no avail.  I've had trouble with this on other programs -- this one seems more promising with ability to upload the cookie file itself.  My log is pretty much same as before when trying:

2021-09-12T02:32:46.504Z ERROR: Error during parsing: Error: Command failed with exit code 1: /app/node_modules/youtube-dl/bin/youtube-dl --dump-json -- https://www.discoveryplus.com/video/[VideoName]

2021-09-12T02:32:46.504Z ERROR: failed to parse for urls starting with https://www.discoveryplus.com/video/[VideoName]

2021-09-12T02:32:47.237Z ERROR: Error while retrieving info on video with URL https://www.discoveryplus.com/video/[VideoName] with the following message: Error: Command failed with exit code 1: /app/node_modules/youtube-dl/bin/youtube-dl --dump-json -f best -o video/%(title)s.mp4 --write-info-json --print-json --write-thumbnail -- https://www.discoveryplus.com/video/[VideoName]

2021-09-12T02:32:47.237Z ERROR: ERROR: This video is only available for registered users. You may want to use --cookies.
2021-09-12T02:32:48.177Z ERROR: ERROR: This video is only available for registered users. You may want to use --cookies

I'll see if @Tzahi12345 has any suggestions.  Thanks for your time!

**Reply by Tzahi12345** [reply]
Looks like most cookie extractors have issues with discovery plus, but I discovered a reddit thread that I think has more info:

https://www.reddit.com/r/youtubedl/comments/lxsns5/discovery_plus_extractor/guset3r/

I think you'll have to use [this extension](https://addons.mozilla.org/en-US/firefox/addon/export-cookies-txt/).

**Reply by Hawkins1212** [reply]
Thanks for the reply.  I actually was already using that extension to pull the cookies.txt file.  I followed instructions, re-pulled the cookies file, imported using Advanced -- Set Cookies, and I still get the same error in the logs.

Essentially, I log on to discovery plus.  Export the cookies.txt file using the "Export cookies" extension.  Go to YoutubeDL-material and "Advanced".  I check "Use Cookies", then click "Set Coookies" and upload my cookies.txt file.  Then I copy the URL from discovery plus show and paste into URL under the "Home" menu.  Download fails.  Is there some reason that setting cookies in the "Advanced" file isn't actually setting cookies using "--cookies?"

**Reply by Tzahi12345** [reply]
If you enable verbose logging (I believe it required a server restart) you can see exactly what args are being used. It should be using the cookies file.

**Comment by Hawkins1212** [comment]
Well I am officially stupid :)  I was doing all this "full screen" and failed to see the "Save" button at the bottom.  I guess I assumed settings auto-saved.  Once I found that, it all seemed to work.  Thanks!!

**Reply by GlassedSilver** [reply]
Hooray! Glad it worked out well for you then! Have fun using the app then and feel free to fire away whenever something else comes across! :)

**Reply by Hawkins1212** [reply]
I will say, it seems I am having some issues.  I download and it gets so far and just stops....I cancel, delete source files, and restart -- it goes a little further and then stops again.  Is there something special when DLing large files (ie. 45 min long or so) i need to be doing?  The furthest I am getting 1.2 GB currently.

**Reply by GlassedSilver** [reply]
Could be it takes a good while until it actually completes. Maybe #420 will take care of it. If the file actually never completes (yt-dl is sussed to be so slow sometimes it can go as low as 70kbyte/s, I found that it can help to just restart the container of YoutubeDL-Material gracefully)

If that doesn't do it, feel free to create an issue for it separately so we can track the issue better and provide better help. :) If it does help, let me know in this thread :) Cheers and hope it works!
